lxd/api.go

```diff
@@ -71,7 +72,24 @@ func restServer(d *Daemon) *http.Server {
 	uiEnabled := uiPath != "" && shared.PathExists(uiPath)
 	if uiEnabled {
 		uiHTTPDir := uiHTTPDir{http.Dir(uiPath)}
-		mux.PathPrefix("/ui/").Handler(http.StripPrefix("/ui/", http.FileServer(uiHTTPDir)))
+
+		// Serve the LXD user interface.
+		uiHandler := http.StripPrefix("/ui/", http.FileServer(uiHTTPDir))
+
+		// Set security headers
+		uiHandlerWithSecurity := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Permissions-Policy", "interest-cohort=()")
+			w.Header().Set("X-Content-Type-Options", "nosniff")
+			w.Header().Set("X-Frame-Options", "SAMEORIGIN")
+			w.Header().Set("Content-Security-Policy", "default-src 'self'; script-src 'self'; object-src 'none'")
```

Comment:
@gabrielmougard where did the requirement for the CSP header come from btw, I don't see it in the original issue?

Reply:
in @simondeziel suggestion:

> `X-Xss-Protection` seems like a deprecated/to be avoided header nowadays: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection

There is a a suggestion to use CSP instead